iac-extensions/docker/src/test/java/org/sonar/iac/docker/utils/ArgumentUtilsTest.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * SonarQube IaC Plugin
+ * Copyright (C) 2021-2022 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.iac.docker.utils;
+
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+import org.sonar.iac.docker.parser.grammar.DockerLexicalGrammar;
+import org.sonar.iac.docker.tree.api.Argument;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.sonar.iac.docker.tree.impl.DockerTestUtils.parse;
+
+class ArgumentUtilsTest {
+
+  @ParameterizedTest
+  @CsvSource({
+    "foo",
+    "\"foo\"",
+    "fo\"o\"",
+    "f\"o\"o",
+    "f'oo'",
+    "f'o'\"o\""
+  })
+  void resolveArgument(String input) {
+    Argument argument = parseArgument(input);
+    assertThat(ArgumentUtils.resolve(argument)).extracting(ArgumentUtils.ArgumentResolution::value)
+      .isNotNull()
+      .isEqualTo("foo");
+  }
```

Comment:
We are missing the use case to show that variable are not replaced in single quote string : `'$foo'`

Reply:
Unfortunately, I'm not able to escape the single quotes correctly. The CsvSource annotation is cropping them all the time.